discord/client.py

```diff
@@ -754,9 +754,7 @@ def allowed_mentions(self):
 
     @allowed_mentions.setter
     def allowed_mentions(self, value):
-        if value is None:
-            self._connection.allowed_mentions = value
-        elif isinstance(value, AllowedMentions):
+        if value is None and isinstance(value, AllowedMentions):
```

Comment:
This condition is now never true.
```suggestion
        if value is None or isinstance(value, AllowedMentions):
```